ccl/changefeedccl: TestChangefeedNemeses failed

ccl/changefeedccl.TestChangefeedNemeses [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20037082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20037082?buildTab=artifacts#/) on master @ [f7b60e4c998f715a4f2d2165c1504fa234414480](https://github.com/cockroachdb/cockroach/commits/f7b60e4c998f715a4f2d2165c1504fa234414480):


```
=== RUN   TestChangefeedNemeses
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a2df272b68db5af56229d82d0d6ebe3f/logTestChangefeedNemeses2227894812
    test_log_scope.go:76: use -show-logs to present logs inline
    nemeses_test.go:71: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a2df272b68db5af56229d82d0d6ebe3f/logTestChangefeedNemeses2227894812
--- FAIL: TestChangefeedNemeses (6.50s)
=== RUN   TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true
--- FAIL: TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true (6.50s)
=== RUN   TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true/kafka
    helpers_test.go:1241: making server as system tenant
    helpers_test.go:1317: making kafka feed factory
    nemeses_test.go:33: random seed: -6590589489860162966
    nemeses_test.go:40: using legacy schema changer
    testfeed_test.go:325: creating external connection
    testfeed_test.go:328: ran create external connection
    nemeses_test.go:47: pq: schema changes are disallowed on table "fprint" because it is locked
--- FAIL: TestChangefeedNemeses/nemeses_options==fp_validator=true,sql_smith=true/kafka (6.47s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedNemeses.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/nemeses_test.go
+++ b/pkg/ccl/changefeedccl/nemeses_test.go
@@ -38,8 +38,11 @@ func TestChangefeedNemeses(t *testing.T) {
 			withLegacySchemaChanger := nop.EnableSQLSmith || rng.Float32() < 0.1
 			if withLegacySchemaChanger {
 				t.Log("using legacy schema changer")
+				sqlDB.Exec(t, "SET create_table_with_schema_locked=false")
 				sqlDB.Exec(t, "SET use_declarative_schema_changer='off'")
-				sqlDB.Exec(t, "SET CLUSTER SETTING  sql.defaults.use_declarative_schema_changer='off'")
+				sqlDB.Exec(t, "SET CLUSTER SETTING sql.defaults.use_declarative_schema_changer='off'")
+				sqlDB.Exec(t, "SET CLUSTER SETTING sql.defaults.create_table_with_schema_locked='false'")
+
 			}
 			v, err := cdctest.RunNemesis(f, s.DB, t.Name(), withLegacySchemaChanger, rng, nop)
 			if err != nil {

--- a/pkg/cmd/roachtest/tests/follower_reads.go
+++ b/pkg/cmd/roachtest/tests/follower_reads.go
@@ -527,6 +527,13 @@ func initFollowerReadsDB(
 		}
 	}
 
+	// Disable schema_locked within this since it will modify locality on
+	// tables.
+	_, err = db.ExecContext(ctx, "SET create_table_with_schema_locked=false")
+	require.NoError(t, err)
+	_, err = db.ExecContext(ctx, "ALTER ROLE ALL SET create_table_with_schema_locked=false")
+	require.NoError(t, err)
+
 	// Create a multi-region database and table.
 	_, err = db.ExecContext(ctx, `CREATE DATABASE mr_db`)
 	require.NoError(t, err)

--- a/pkg/cmd/roachtest/tests/sqlsmith.go
+++ b/pkg/cmd/roachtest/tests/sqlsmith.go
@@ -420,12 +420,15 @@ func setupMultiRegionDatabase(t test.Test, conn *gosql.DB, rnd *rand.Rand, logSt
 	}
 
 	for _, table := range tables {
+		// Locality changes can only be made if schema_locked is toggled.
+		execStmt(fmt.Sprintf(`ALTER TABLE %s SET (schema_locked=false);`, table.String()))
 		// Maybe change the locality of the table.
 		if val := rnd.Intn(3); val == 0 {
 			execStmt(fmt.Sprintf(`ALTER TABLE %s SET LOCALITY REGIONAL BY ROW;`, table.String()))
 		} else if val == 1 {
 			execStmt(fmt.Sprintf(`ALTER TABLE %s SET LOCALITY GLOBAL;`, table.String()))
 		}
+		execStmt(fmt.Sprintf(`ALTER TABLE %s SET (schema_locked=true);`, table.String()))
 		// Else keep the locality as REGIONAL BY TABLE.
 	}
 }

